src/engine/Service.cpp

```diff
@@ -625,5 +635,14 @@ void Service::precomputeSiblingResult(std::shared_ptr<Operation> left,
 // _____________________________________________________________________________
 std::unique_ptr<Operation> Service::cloneImpl() const {
   return std::make_unique<Service>(_executionContext, parsedServiceClause_,
-                                   getResultFunction_);
+                                   networkFunctions_);
+}
+
+// _____________________________________________________________________________
+void Service::handleChildRuntimeInfoUpdate(const std::string& msg) {
+  try {
+    childRuntimeInformation_ =
+        std::make_shared<RuntimeInformation>(nlohmann::json::parse(msg));
+  } catch (const nlohmann::json::parse_error&) {
+  }
```

Comment:
This may technically only be called while
the Service is running (e.g. the lazy generator is active) to avoid data races.

For example you can set an atomic flag to false before each `co_yield`, and set it to true at the start + after the co_yields.
When the flag is set to true, you can safely write the runtime information + also call the `signalUpdate` function, otherwise you just have to cache the msg somewhere and send it once the flag is set the next time etc.